todos-gateway.js

```diff
@@ -0,0 +1,23 @@
+const uuid = require('uuid/v4')
+
+module.exports = function todosGateway(collection) {
+  return {
+    async create(newTodo) {
+      const todo = Object.assign({ id: uuid() }, newTodo)
+      await collection.insertOne(todo)
+      return todo
+    },
+    async findById(id) {
+      const found = await collection.find({ id }).toArray()
```

Comment:
Only one should have a matching `id`, so this function shouldn't return an `Array`.